transmission/transmission_test.go

```diff
@@ -820,6 +823,100 @@ func TestFireBatchWithBrokenFirstEvent(t *testing.T) {
 	})
 }
 
+type fakeBatch struct {
+	block <-chan struct{}
+	send  chan<- []interface{}
+	items []interface{}
+}
+
+func (fb *fakeBatch) Add(item interface{}) {
+	fb.items = append(fb.items, item)
+}
+
+func (fb *fakeBatch) Fire(notifier muster.Notifier) {
+	defer notifier.Done()
+	<-fb.block
+	fb.send <- fb.items
+}
+
+func TestHoneycombTransmissionFlush(t *testing.T) {
+	ev := &Event{
+		Metadata: "adder",
+	}
+
+	t.Run("Flush should send all data that has been Added", func(t *testing.T) {
+		w := new(Honeycomb)
+		w.MaxBatchSize = 1000
+		w.PendingWorkCapacity = 1
+		block := make(chan struct{})
+		close(block) // don't block
+		sendChan := make(chan []interface{})
+		b := &fakeBatch{
+			send:  sendChan,
+			block: block,
+		}
+
+		batchCount := int32(0)
+		w.batchMaker = func() muster.Batch {
+			t.Logf("creating batch %d", atomic.LoadInt32(&batchCount))
+			if atomic.CompareAndSwapInt32(&batchCount, 0, 1) {
+				return b
+			}
+			return &fakeBatch{}
+			// We want to be sure that the data we enqueue is flushed in the batch we
+			// expect. By not having s send channel on subsequent batches, we'll
+			// panic if we flush to other batches.
+		}
+
+		if err := w.Start(); err != nil {
+			t.Error("unable to start", err)
+		}
+		defer w.Stop()
+		w.Add(ev)
+		go func() {
+			if err := w.Flush(); err != nil {
+				t.Error("unable to flush", err)
+			}
+		}()
+		items := <-sendChan
+		testEquals(t, len(items), 1, "should be exactly one item")
+		testEquals(t, items[0], ev, "one item should be the event we added")
+	})
```

Comment:
maybe it's just too early in the morning, but I'm not putting together all the pieces to understand what the various constraints are that this test is exercising. Could you add a comment at the beginning of the test describing at a higher level the specific behavior it's exercising?

Reply:
This test flow is pretty confusing. I wanted to ensure that `Flush` was causing `muster` to call `Fire` on the batch, so I had to use this channel to signal that happening. I added a comment for this test and for `fakeBatch` Let me know if those have helped.